smart-connector-rest-server/src/main/java/eu/knowledge/engine/rest/api/impl/ProactiveApiServiceImpl.java

```diff
@@ -46,6 +46,126 @@ public class ProactiveApiServiceImpl {
 
 	private RestKnowledgeBaseManager manager = RestKnowledgeBaseManager.newInstance();
 
+	@POST
+	@Path("/askwithgaps")
+	@Consumes({ "application/json; charset=UTF-8" })
+	@Produces({ "application/json; charset=UTF-8", "text/plain; charset=UTF-8" })
+	public void scAskWithGapsPost(
```

Comment:
Can we, instead of a separate REST operation for the knowledge gaps, include the knowledge gaps in the `/ask` operation when they are available? And leave them out altogether when they are not available?

Reply:
Yes, but this is work in progress....the current exercise was to have the knowledge gaps being nicely provided via the java API.